core/src/main/java/io/fabric8/maven/core/util/MavenUtil.java

```diff
@@ -246,4 +251,56 @@ public static void createArchive(File sourceDir, File destinationFile, JarArchiv
             throw new MojoExecutionException("Failed to create archive " + destinationFile + ": " + e, e);
         }
     }
+
+    /**
+     * Returns the version from the list of pre-configured versions of common groupId/artifact pairs
+     */
+    public static String getVersion(String groupId, String artifactId) {
+        String key = "" + groupId + "/" + artifactId;
+        Map<String, String> map = getGroupArtifactVersionMap();
+        String version = map.get(key);
+        if (version == null) {
+            LOG.warn("Could not find the version for groupId: " + groupId + " artifactId: " + artifactId + " in: " + map);
+        }
+        return version;
+    }
+
+    /**
+     * Returns the version from the list of pre-configured versions of common groupId/artifact pairs
+     */
+    public static String getVersion(String groupId, String artifactId, String defaultVersion) {
```

Comment:
for simplicity reasons I wouldn't allow a `defaultVersion`. If the version could not be found in the properties instead an internal error exception should be thrown to show prominently that something internally is wrong. I think its fair to assume that the version from `versions.properties` must be *always* loadable.

That also avoids to be forced to update the version in multiple places.